zmon_aws_agent/aws.py

```diff
@@ -783,3 +783,44 @@ def get_limits(region, acc, apps, elbs):
     entity.update(limits)
 
     return entity
+
+
+def get_sqs_queues(region, acc):
+    sqs_queues = []
+
+    try:
+        sqs_client = boto3.client('sqs', region_name=region)
+        list_queues_response = call_and_retry(lambda: sqs_client.list_queues())
+        for queue_url in list_queues_response['QueueUrls']:
+            try:
+                attributes_response = call_and_retry(
+                    lambda: sqs_client.get_queue_attributes(QueueUrl=queue_url, AttributeNames=['All']))
+                attributes = attributes_response['Attributes']
+                queue_arn = attributes['QueueArn']
+                match = re.match(r"^arn:aws:sqs:(?P<region>.+):(?P<account>.+):(?P<name>.+)$", queue_arn)
+                if match:
+                    queue_name = match.group('name')
+                else:
+                    raise ValueError('Illegal SQS queue ARN: "{}"'.format(queue_arn))
+
+                sqs_queues.append({
+                    'id': entity_id('sqs-{}[{}:{}]'.format(queue_name, acc, region)),
+                    'created_by': 'agent',
+                    'infrastructure_account': '{}'.format(acc),
+                    'region': region,
+                    'type': 'aws_sqs',
+                    'name': queue_name,
+                    'url': queue_url,
+                    'arn': queue_arn,
+                    'message_retention_period_seconds': int(attributes.get('MessageRetentionPeriod', "0")),
+                    'maximum_message_size_bytes': int(attributes.get('MaximumMessageSize', "0")),
+                    'receive_messages_wait_time_seconds': int(attributes.get('ReceiveMessageWaitTimeSeconds', "0")),
+                    'delay_seconds': int(attributes.get('DelaySeconds', "0")),
+                    'visibility_timeout_seconds': int(attributes.get('VisibilityTimeout', "0"))
+                })
```

Comment:
I'd prefer to follow the defaults to coalesce missing values. For some of these, 0 seem to be wrong.
MaximumMessageSize: 262,144
MessageRetentionPeriod: 345,600 (4 days)
VisibilityTimeout: 30
The others default to 0.
I'd expect to have some visibility on the RedrivePolicy since this could change the reaction on about to expire msgs (having a DLQ could change the alerting strategy)